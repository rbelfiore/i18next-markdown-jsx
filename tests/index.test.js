import i18next from 'i18next'
import ReactDOM from 'react-dom'

import markdownJsx from '../src/index'

describe('markdown jsx', () => {

  const root = document.body.appendChild(document.createElement('div'));
  function render (jsx) { return ReactDOM.render(jsx, root); }

  afterEach(() => ReactDOM.unmountComponentAtNode(root));

  const i18n = i18next
    .use(markdownJsx)
    .init({
      lng: 'en',
      defaultNS: 'ns1',
      resources: {
        en: {
          ns1: {
            key1: '**Test 1**'
          }
        }
      }
    });

  it('correctly translates key', () => {
    expect(i18n.t('key1')).toEqual('**Test 1**')
  })

  it('correctly translates key with markdown', () => {
    render(i18n.t('key1', {postProcess:'markdown-jsx'}))
    expect(root.innerHTML).toMatchSnapshot()
  })
});