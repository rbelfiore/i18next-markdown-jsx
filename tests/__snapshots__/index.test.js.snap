// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`markdown jsx correctly translates key with markdown 1`] = `"<strong>Test 1</strong>"`;
